jib-core/src/main/java/com/google/cloud/tools/jib/image/Image.java

```diff
@@ -125,8 +127,8 @@
      * @param exposedPorts the list of exposed ports to add
      * @return this
      */
-    public Builder<T> setExposedPorts(@Nullable List<Port> exposedPorts) {
-      this.exposedPorts = (exposedPorts == null) ? null : ImmutableList.copyOf(exposedPorts);
+    public Builder<T> setExposedPorts(@Nullable Set<Port> exposedPorts) {
```

Comment:
Just saying here, I think eventually this and `setVolumes()` would need to change to `add...()`, like other methods such as `addLabels()`, `addEnvironment()` so that we inherit the base settings and only add new values.

Reply:
Yeah, was gonna do this when I add/change the inheritance stuff. Is the plan to replace the set methods with add methods, or just add new add methods alongside the set ones?

Reply:
IMHO, we should keep set and add.  The build plugins should strive to mimic Docker behaviour, but the library should keep an open mind.

Reply:
Isn't the `Image` class basically internal? (It's not under `com.google.cloud.tools.jib.api`.) Looks like we won't be using `setExportedPorts()`.

Reply:
And since it's the builder method, and you always start with an empty port, even if some library user decided to use it directly. That is, `addExportedPorts()` will effectively be `setExportedPorts()` when called for the first time.